runtime/doc/diagnostic.txt

```diff
@@ -0,0 +1,524 @@
+variety of sources, such as linters or LSP servers. The diagnostic framework
+is an extension to existing error handling functionality such as the
+|quickfix| list.
+
+                                      Type |gO| to see the table of contents.
+
+==============================================================================
+QUICKSTART                                              *diagnostic-quickstart*
+
+Anything that reports diagnostics is referred to below as a "diagnostic
+producer". Diagnostic producers need only follow a few simple steps to
+report diagnostics:
+
+1. Create a namespace |nvim_create_namespace()|. Note that the namespace must
+   have a name. Anonymous namespaces WILL NOT WORK.
+2. (Optional) Configure options for the diagnostic namespace
+   |vim.diagnostic.config()|.
+3. Generate diagnostics.
+4. Set the diagnostics for the buffer |vim.diagnostic.set()|.
+5. Repeat from step 3.
+
+Generally speaking, the API is split between functions meant to be used by
+diagnostic producers and those meant for diagnostic consumers (i.e. end users
+who want to read and view the diagnostics for a buffer).  The APIs for
+producers require a {namespace} as their first argument, while those for
+consumers generally do not require a namespace (though often one may be
+optionally supplied).  A good rule of thumb is that if a method is meant to
+modify the diagnostics for a buffer (e.g. |vim.diagnostic.set()|) then it
+requires a namespace.
+
+							*diagnostic-structure*
+A diagnostic is a Lua table with the following keys:
+
+	lnum: The starting line of the diagnostic
+	end_lnum: The final line of the diagnostic
+	col: The starting column of the diagnostic
+	end_col: The final column of the diagnostic
+	severity: The severity of the diagnostic |vim.diagnostic.severity|
+	message: The diagnostic text
+
+Diagnostics use the same indexing as the rest of the Nvim API (i.e. 0-based
+rows and columns). |api-indexing|
+
+				*vim.diagnostic.severity* *diagnostic-severity*
+The "severity" key in a diagnostic is one of the values defined in
+`vim.diagnostic.severity`:
+
+	vim.diagnostic.severity.ERROR
+	vim.diagnostic.severity.WARN
+	vim.diagnostic.severity.INFO
+	vim.diagnostic.severity.HINT
+
+==============================================================================
+HIGHLIGHTS						*diagnostic-highlights*
+
+All highlights defined for diagnostics begin with `Diagnostic` followed by
+the type of highlight (e.g., `Sign`, `Underline`, etc.) and the severity (e.g.
+`Error`, `Warn`, etc.)
+
+Sign, underline and virtual text highlights (by default) are linked to their
+corresponding DiagnosticDefault highlight.
+
+For example, the default highlighting for |hl-DiagnosticSignError| is
+linked to |hl-DiagnosticDefaultError|. To change the default (and
+therefore the linked highlights), use the |:highlight| command: >
+
+    highlight DiagnosticDefaultError guifg="BrightRed"
+<
+						*hl-DiagnosticDefaultError*
+DiagnosticDefaultError
```

Comment:
since we are renaming these anyway let's remove "Default" from the highlight names. Don't want this to proliferate. It is redundant and unnecessary. All highlights may (or may not) have "default" forms.
```suggestion
DiagnosticError
```